### PR TITLE
Azure load testing API version 2024-07-01-preview

### PR DESCRIPTION
**Pull request by venkatr21**
## API Info: The Basics

1. Supporting pass fail criteria based on server-side metrics for Azure load testing 
2. Displaying the VUserHours for the test run, which is an estimate of billing consumption of the test run
3. Scheduling test run changes for Azure load testing which is a new set of API called triggers

### Change Scope

This PR adds a 
1. `PassFailServerCriteria` inside which has fields like resource id of the server, metric namespace, metric name, aggregation, threshold values
2. Identity reference for metrics which is the identity used to fetch metrics from the customer's server
3. API changes for newly introduced trigger API for scheduling test run
4. Add VUH details to the test run

### CODE REVIEW

File: specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json [hunk 1]
```diff
@@ -0,0 +1,6572 @@
...
+    "ListTestRunQueryParams.testId": {
+      "name": "testId",
+      "in": "query",
+      "description": "Unique name of an existing load test.",
+      "required": false,
+      "type": "string",
+      "x-ms-parameter-location": "method"
+    },
+    "ListTriggerQueryParams.createdFrom": {
+      "name": "createdFrom",
```
Comment (mikekistler) [inline specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json:6417]: I think this parameter should be renamed for consistency with the existing `createdDateStartTime` parameter of the `ListTestProfileRun` operation.
```suggestion
      "name": "createdDateStartTime",
```

File: specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json [hunk 2]
```diff
@@ -0,0 +1,6572 @@
...
+      "in": "query",
+      "description": "Start DateTime of trigger creation time filter range. (RFC 3339 literal format)",
+      "required": false,
+      "type": "string",
+      "format": "date-time",
+      "x-ms-parameter-location": "method"
+    },
+    "ListTriggerQueryParams.createdTo": {
+      "name": "createdTo",
```
Comment (mikekistler) [inline specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json:6426]: Another rename for consistency
```suggestion
      "name": "createdDateEndTime",
```

**Review by mikekistler — APPROVED** [review]
Looks good. 👍 

I noticed a minor inconsistency in date-time parameter naming that I flagged in my comments. That would be good to fix but not absolutely necessary for preview.

File: specification/loadtestservice/LoadTestService/models.trigger.tsp [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+  @doc("Filter triggers based on a comma separated list of states.")
+  @query
+  states?: TriggerState;
+
+  @doc("Start DateTime of trigger creation time filter range. (RFC 3339 literal format)")
+  @query
+  createdFrom?: utcDateTime;
+
+  @doc("End DateTime of trigger creation time filter range. (RFC 3339 literal format)")
+  @query
```
Comment (mitsha-microsoft) [inline specification/loadtestservice/LoadTestService/models.trigger.tsp:271]: @venkatr21/ @Himanshu49 These parameters dont exist. Replace with `lastModifiedStartTime` and `lastModifiedEndTime` . Sorry missed this in review and it is in line with Mike's suggestion

Comment (mitsha-microsoft) [reply specification/loadtestservice/LoadTestService/models.trigger.tsp:275]: Fixed it

**Review by mitsha-microsoft — APPROVED** [review]

File: specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json (hunk 1 above)
Comment (venkatr21) [reply specification/loadtestservice/data-plane/Microsoft.LoadTestService/preview/2024-07-01-preview/loadtestservice.json]: @mikekistler fixed it

File: specification/loadtestservice/LoadTestService/models.trigger.tsp [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+  @doc("Refers to Saturday.")
+  Saturday: "Saturday",
+}
+
+@added(APIVersions.v2024_07_01_preview)
+@doc("Type of a trigger.")
+union TriggerType {
+  string,
+  ScheduleTestsTrigger: "ScheduleTestsTrigger",
```
Comment (weidongxu-microsoft) [inline specification/loadtestservice/LoadTestService/models.trigger.tsp:86]: nit, `@doc` on the enum value.

File: specification/loadtestservice/LoadTestService/models.trigger.tsp [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+}
+
+@added(APIVersions.v2024_07_01_preview)
+@discriminator("frequency")
+@doc("Recurrence model.")
+model Recurrence {
+  @doc("Frequency of the recurrence.")
+  frequency: Frequency;
+
+  @doc("Recurrence end model. You can specify the end either by providing a numberOfOccurrences (which will end the recurrence after the specified number of occurrences) or by providing an endDate (which will end the recurrence after the specified date). If neither value is provided, the recurrence will continue until it is manually ended. However, if both values are provided, an error will be thrown.")
```
Comment (weidongxu-microsoft) [inline specification/loadtestservice/LoadTestService/models.trigger.tsp:251]: nit, `endDateTime`?

File: specification/loadtestservice/LoadTestService/models.trigger.tsp [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+  recurrenceEnd?: RecurrenceEnd;
+}
+
+@added(APIVersions.v2024_07_01_preview)
+@doc("Parameters for list trigger operation")
+model ListTriggerQueryParams {
+  @doc("Search based on triggers associated with the provided test ids.")
+  @query
+  testIds?: string;
```
Comment (weidongxu-microsoft) [inline specification/loadtestservice/LoadTestService/models.trigger.tsp:260]: Why `testIds` plural is a string?

Should it be a csv list or exlode params?

File: specification/loadtestservice/LoadTestService/models.trigger.tsp [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+
+  @doc("Filter triggers based on a comma separated list of states.")
+  @query
+  states?: TriggerState;
```
Comment (weidongxu-microsoft) [inline specification/loadtestservice/LoadTestService/models.trigger.tsp:264]: Same here. It should be `TriggerState[]` as csv.

**Review by weidongxu-microsoft — APPROVED** [review]

File: specification/loadtestservice/LoadTestService/models.trigger.tsp (hunk 6 above)
Comment (mitsha-microsoft) [reply specification/loadtestservice/LoadTestService/models.trigger.tsp:262]: Resolving as discussed offline, we will move forward with this and clean it up later

File: specification/loadtestservice/LoadTestService/models.trigger.tsp (hunk 5 above)
Comment (mitsha-microsoft) [reply specification/loadtestservice/LoadTestService/models.trigger.tsp]: Fixed

File: specification/loadtestservice/LoadTestService/models.trigger.tsp (hunk 4 above)
Comment (mitsha-microsoft) [reply specification/loadtestservice/LoadTestService/models.trigger.tsp:88]: Fixed

**Pull request merged**